--- lash-0.5.3/m4/readline.m4.orig	2007-01-13 22:53:42.000000000 +0100
+++ lash-0.5.3/m4/readline.m4	2007-06-25 21:34:42.745732741 +0200
@@ -27,6 +27,7 @@
       LIBS="$ORIG_LIBS"
     else
 	  READLINE_LIBS=$LIBS
+	  LIBS="$ORIG_LIBS"
 	fi
   ])
 
@@ -37,7 +38,10 @@
     AC_CACHE_CHECK([whether readline supports history],
                    vl_cv_lib_readline_history, [
       vl_cv_lib_readline_history="no"
+      ORIG_LIBS="$LIBS"
+      LIBS="$LIBS $READLINE_LIBS"
       AC_TRY_LINK_FUNC(add_history, vl_cv_lib_readline_history="yes")
+      LIBS="$ORIG_LIBS"
     ])
     if test "$vl_cv_lib_readline_history" = "yes"; then
       AC_DEFINE(HAVE_READLINE_HISTORY, 1,
